R/fread.R

```diff
@@ -36,6 +36,33 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
   )
   nThread=as.integer(nThread)
   stopifnot(nThread>=1L)
+
+  is_url = function(x) any(startsWith(x, c("http://", "https://", "ftp://", "ftps://", "file://")))
```

Comment:
I glanced at our backport of `startsWith` to support R prior to 3.3.0 (see utils.R) and it doesn't look like it supports a set like that. If so, our backport needs a tweak then.

Reply:
I changed the usage here, before fiddling with the backport. This also seems okayish since supported set of URLs will probably not change.